elements/pl-order-blocks/dag_checker.py

```diff
@@ -45,27 +46,92 @@ def check_grouping(order: list[str], group_belonging: Mapping[str, Optional[int]
 
     top_sort_correctness = check_topological_sorting(order, graph)
     grouping_correctness = check_grouping(order, group_belonging)
 
-    correctness = top_sort_correctness if top_sort_correctness < grouping_correctness else grouping_correctness
-    first_wrong = -1 if correctness == len(order) else correctness
-    return correctness, first_wrong
+    return top_sort_correctness if top_sort_correctness < grouping_correctness else grouping_correctness
+
+
+def is_vertex_cover(G: nx.DiGraph, vertex_cover: Iterable):
+    """ this function from
+    https://docs.ocean.dwavesys.com/projects/dwave-networkx/en/latest/_modules/dwave_networkx/algorithms/cover.html#is_vertex_cover
+    """
+    cover = set(vertex_cover)
+    return all(u in cover or v in cover for u, v in G.edges)
+
+
+def lcs_partial_credit(order: list[str], depends_graph: Mapping[str, list[str]]) -> int:
+    """Computes the number of edits required to change the student solution into a correct solution using
+    largest common subsequence edit distance (allows only additions and deletions, not replacing).
+    The naive solution would be to enumerate all topological sorts, then get the edit distance to each of them,
+    but this would be too slow. Instead, our algorithm is as follows:
+        1. Remove all distractors from the student solution
+        2. Construct the 'problematic subgraph'. We do this by finding the subset of nodes in the
+        graph that have some 'problematic' relation, meaning that there is a node before it in the
+        student solution that should be after it, or one after it that must be before. Thus the
+        edge (u,v) is in the problematic subgraph if there is an edge from u to v in the transitive
+        closure of the dependency graph, but v appears before u in the student solution.
+        3. Find the Minimum Vertex Cover (MVC) of the problematic subgraph. This tells us the
+        minimum number of nodes that must be deleted to get rid of all problematic relations.
+        4. Once we know the minimum required deletions, you may simply add nodes to the student
+        solution until it is the correct solution, so you can directly calculate the edit distance.
+    :param order: the block ordering given by the student
+    :param depends_graph: The dependency graph between blocks specified in the question
+    :return: edit distance from the student submission to some correct solution
+    """
+    graph = dag_to_nx(depends_graph)
+    trans_clos = nx.algorithms.dag.transitive_closure(graph)
+
+    seen = set()
+    subgraph = nx.DiGraph()
+    distractors = 0
+    for node in order:
+        if node is None:
+            distractors += 1
+            continue
+
+        for node2 in seen:
+            if trans_clos.has_edge(node, node2):
+                subgraph.add_edge(node, node2)
+
+        seen.add(node)
+
+    if len(subgraph) == 0:
```

Comment:
Nit: I'd rather use this instead for extra clarity - I had to go and look up how `len(nx.DiGraph())` was defined, but this should be unambiguous.
```suggestion
    if subgraph.number_of_nodes() == 0:
```

Reply:
sounds good.